test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -609,11 +610,18 @@ func waitForReplicas(ctx context.Context, deploymentName, namespace string, cs c
 	}
 }
 
+func interrupted(err error) bool {
+	return wait.Interrupted(err) || strings.Contains(err.Error(), "would exceed context deadline")
+}
+
 func ensureDesiredReplicasInRange(ctx context.Context, deploymentName, namespace string, cs clientset.Interface, minDesiredReplicas, maxDesiredReplicas int, timeout time.Duration) {
 	interval := 60 * time.Second
 	err := wait.PollUntilContextTimeout(ctx, interval, timeout, true, func(ctx context.Context) (bool, error) {
 		deployment, err := cs.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
 		if err != nil {
+			if interrupted(err) {
+				return true, err
+			}
 			framework.Failf("Failed to get replication controller %s: %v", deployment, err)
```

Comment:
the problem us here, we should not use a failf inside the loop and just return the error